src/Command/CommandBase.php

```diff
@@ -1147,6 +1147,98 @@ protected function convertEnvironmentAliasToUuid(InputInterface $input): void {
     }
   }
 
+  /**
+   * @param $cloud_environment
+   *
+   * @return string
+   */
+  public static function getSiteGroupFromSshUrl($cloud_environment): string {
+    $ssh_url_parts = explode('.', $cloud_environment->sshUrl);
+    $sitegroup = reset($ssh_url_parts);
+
+    return $sitegroup;
+  }
+
+  /**
+   * @param $cloud_environment
+   *
+   * @return bool
+   */
+  protected function isAcsfEnv($cloud_environment): bool {
+    if (strpos($cloud_environment->sshUrl, 'enterprise-g1') !== FALSE) {
+      return TRUE;
+    }
+    foreach ($cloud_environment->domains as $domain) {
+      if (strpos($domain, 'acsitefactory') !== FALSE) {
+        return TRUE;
+      }
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * @param \AcquiaCloudApi\Response\EnvironmentResponse $cloud_environment
+   *
+   * @return array
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function getAcsfSites($cloud_environment): array {
+    $sitegroup = self::getSiteGroupFromSshUrl($cloud_environment);
+    $command = ['cat', "/var/www/site-php/$sitegroup.{$cloud_environment->name}/multisite-config.json"];
+    $process = $this->sshHelper->executeCommand($cloud_environment, $command, FALSE);
+    if ($process->isSuccessful()) {
+      return json_decode($process->getOutput(), TRUE);
+    }
+    throw new AcquiaCliException("Could not get ACSF sites");
+  }
+
+  /**
+   * @param \AcquiaCloudApi\Response\EnvironmentResponse $cloud_environment
+   *
+   * @return array
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function getCloudSites($cloud_environment): array {
+    $sitegroup = self::getSiteGroupFromSshUrl($cloud_environment);
+    $command = ['ls', "/mnt/files/$sitegroup.{$cloud_environment->name}/sites"];
```

Comment:
Have you verified that this works on both CDEs and ACE sites?

Reply:
Yep I've tested on both CDE and ACE